src/btx/data/utils.py

```diff
@@ -56,3 +56,49 @@ def map(self, sample: dict[str, object]) -> dict[str, object]:
         sample["scale"] = np.array([scale_x, scale_y])
 
         return sample
+
+@beartype.beartype
+@dataclasses.dataclass(frozen=True)
+class GaussianHeatmap(grain.transforms.Map):
+    size: int = 256
+    """Image size in pixels (used if we can't infer from sample['img'])."""
+    sigma: float = 3.0
+    """Standard deviation of the Gaussian in pixels."""
+
+    def map(self, sample: dict[str, object]) -> dict[str, object]:
+        """
+        Reads 'tgt' (keypoint coordinates) and adds:
+          sample['tgt_pixel_probs']: Float[np.ndarray, "points height width"]
+
+        Each channel is a 2D Gaussian centered at the corresponding keypoint,
+        with peak value 1.0 and std-dev = self.sigma (in pixels).
+        """
+        if "img" in sample and isinstance(sample["img"], np.ndarray):
+            H, W = sample["img"].shape[:2]
+        else:
+            H = W = int(self.size)
+        tgt = np.asarray(sample["tgt"], dtype=np.float32)  # shape: (lines, 2, 2)
```

Comment:
sample['tgt'] should already be a dtype float32 np.ndarray. Let's do asserts instead of casting. Something like `assert isinstance(sample['tgt'], np.ndarray)` and `assert sample['tgt'].dtype == np.float32`.